src/salt.ts

```diff
@@ -0,0 +1,15 @@
+// Moved to a separate module for easier mocking in tests.
+import { v4 as uuidv4 } from 'uuid';
+
+/**
+ * Generate a random salt for use in obfuscation. The returned value is guaranteed to be a valid
+ * UTF-8 string and have enough entropy for obfuscations. Other than that, the output format is not
+ * defined.
+ *
+ * @internal
+ */
+export function generateSalt() {
```

Comment:
actually found that SDK was not generating salt and getPrecomputedConfiguration was asking user to pass salt (which is a bad idea as I guess it was never passed)